lms/djangoapps/gdpr_user_retirement/views.py

```diff
@@ -0,0 +1,83 @@
+"""
+An API for retiring user accounts.
+"""
+import logging
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from rest_framework import permissions, status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from six import text_type
+from social_django.models import UserSocialAuth
+from student.models import AccountRecovery, Registration, get_retired_email_by_email
+from openedx.core.djangolib.oauth2_retirement_utils import retire_dot_oauth2_models
+from openedx.core.djangoapps.user_api.models import UserRetirementStatus
+from openedx.core.djangoapps.user_api.accounts.permissions import CanRetireUser
+
+log = logging.getLogger(__name__)
+
+
+class GDPRUsersRetirementView(APIView):
+    """
+    **Use Case**
+
+        Implementation for GDPR User Retirement API. Creates a retirement request
+        for one or more users.
+
+    **Example Request**
+
+        POST /v1/accounts/gdpr_retire_users {
+            "usernames": "test_user1, test_user2"
+        }
+
+        **POST Parameters**
+
+          A POST request can include the following parameter.
+
+          * usernames: Comma separated strings of usernames that should be retired.
+    """
+    authentication_classes = (JwtAuthentication, )
+    permission_classes = (permissions.IsAuthenticated, CanRetireUser)
```

Comment:
As this is a destructive API I would definitely recommend adding some test cases to ensure that unauthorized users cannot retire other users.

Reply:
I added another test case for unauthorised users.